utils/hare_cov/hare_coverage.py

```diff
@@ -0,0 +1,362 @@
+        """
+        for word in self.inc_kwargs:
+            if dir_name.find(word) > -1:
+                return False
+        return True
+
+    def search_dir(self, dir_path: str) -> List[str]:
+        """
+        Strategy for scanning directories.
+        :param dir_path: directory path
+        :return: List of the file match to given pattern
+        """
+        result_list: List[str] = []
+        for sub_dir, dirs, _ in walk(dir_path):
+            for directory in dirs:
+                directory_path: str = sub_dir + sep + directory
+                if self.omit_dir(directory_path):
+                    continue
+                result_list.extend([ele for ele in
+                                    self.search_dir(directory_path)
+                                    if ele not in result_list])
+                if self.callback(directory):
+                    if directory_path not in result_list:
+                        result_list.append(directory_path)
+        return result_list
+
+
+class Utility:
+    """Here is the utility class which has static utility helper methods."""
+
+    def run_cmd(self, command: str, delimiter: Optional[str] = None,
+                max_split: int = -1) -> None:
+        """
+        Execute command and return the result.
+        :param command: Command in string format
+        :param delimiter: delimiter (default is space)
+        :param max_split: maximum number split
+        :return: tuple(stdout, stderr)
+        """
+        logging.debug(f"Executing Command : {command!a}")
+        process: Popen[bytes] = Popen(
+            command.split(sep=delimiter, maxsplit=max_split),
+            stdout=PIPE,
+            stderr=PIPE)
+
+        out, err = process.communicate()
+        if process.returncode:
+            raise Exception(
+                f'Command {command!a} exited with error code '
+                f'{process.returncode}.\n'
+                f'Command output: {err.decode("utf-8")}')
+
+        logging.debug(f'Output:\n{out.decode("utf-8")}')
+
+    def is_file_system_node_exist(self, dir_path: str) -> bool:
+        """
+        Check for the existence of the file system node.
+        :param dir_path: file or directory path
+        :return: bool
+        """
+        return path.exists(dir_path)
+
+    def get_abs_hare_home_path(self) -> str:
+        """
+        Search for the absolute path from where the script can search for
+        the test and source directory for unit testcase and coverage run.
+        :return: absolute hare home path in string.
+        """
+        absolute_file_path = path.realpath(__file__)
+        pos = absolute_file_path.find(f"utils{sep}hare_cov{sep}"
+                                      "hare_coverage.py")
+        if pos != -1:
+            return absolute_file_path[:pos]
+        else:
+            logging.error('Not able to get relative hare home path.')
+            print('Not able to get relative hare home path.')

```

Comment:
Please don't use print, consider logging.

It is possible to configure the logger to write both to the screen and to the file simultaneously, please check this function as an example: https://github.com/Seagate/cortx-ha/blob/19c6cad22b4a7f2036bc7e292b40383bae4d7fdb/ha/pcswrap/pcswrap/client.py#L208-L215

Reply:
Done